tests/internal/test_serverless.py

```diff
@@ -0,0 +1,52 @@
+import os
+
+import mock
+
+from ddtrace.internal.serverless.mini_agent import maybe_start_serverless_mini_agent
+
+
+@mock.patch("ddtrace.internal.serverless.mini_agent.Popen")
+def test_dont_spawn_mini_agent_if_not_cloud_function(mock_popen):
+    os.environ["DD_MINI_AGENT_PATH"] = "fake_path"
```

Comment:
I tried to use `@pytest.mark.subprocess(env=dict())` and `@run_in_subprocess(env_overrides=dict())` instead of manually setting & unsetting environment variables. 

with `@run_in_subprocess`, the environment variables weren't accessible within the function. with `@pytest.mark.subprocess`, the mock was "undefined" whenever I tried to access it within the function.

Let me know if I'm missing something with using either of the run test in subprocess annotations!